### ISSUE TITLE
Splitting config files

### ISSUE THREAD

**Issue by waghanza** [dependabot/feedback#385]
Hi,

As you know https://github.com/the-benchmarker/web-frameworks/blob/master/.dependabot/config.yml is a very huge config files.

It could be helpful to split configuration into multiple files.

If I may, I think of a solution : having a service plugged as a `github ` or `gitlab` **hook** that store the configuration found in repository (so as an issue could be created if configuration is not valid)

Regards,

**Comment by greysteil** [comment]
Interesting. We definitely buy the "massive file" problem, and are keen to solve it. We've mooted supporting JSON so that jsonnet could be used to generate the file, but that wouldn't solve the problem of it becoming out-of-date when new directories you want to track are added. Fixing that likely needs support for globs.

Can you say a bit more about what you were thinking about with a hook?

**Comment by waghanza** [comment]
sure, the idea behind is that on  **push** (I mean the default branch on a repository track by @dependabot-bot is updated), a `middleware` is called that
+ aggregate / validate configuration files
+ store configuration for this reposiory on **your** side (so as it can be shared by any task)

**Comment by greysteil** [comment]
So basically using an API to create update configs, rather than config files? The API docs [here](https://github.com/dependabot/api-docs) ought to be enough to do that already (I'll add documentation for a DELETE endpoint now.)

**Comment by waghanza** [comment]
what I mean by hook is a **github** webhook trig on each **push**
https://github.com/the-benchmarker/web-frameworks/settings/hooks/new

**Comment by greysteil** [comment]
Or maybe I'm misunderstanding. Dependabot already updates its setup every time a new config file is pushed. What am I missing here?

**Comment by waghanza** [comment]
yeah an API, as you said in https://github.com/dependabot/feedback/issues/385#issuecomment-468316471, but called by github

**Comment by waghanza** [comment]
what I imagine is that `.dependabot/config.yml` should be aggregated/validated/update on each push + stored at **your** side

I mean, having a `.dependabot/config.yml` splitted, with aliases ..., and read on each push to avoid memory leaks, as said https://github.com/dependabot/feedback/issues/70#issuecomment-467495685

**Comment by greysteil** [comment]
Ah, OK. What do you mean by splitted? Sorry for all the questions!

**Comment by waghanza** [comment]
Np for all questions, it's a pleasure to help :heart: 

I mean in separated files. In fact, for me a repository **SHOULD** contains sources files (in `.dependabot/config.d/*` for example), and @dependabot-bot **SHOULD** read a aggregated version of this/those file(s)

**Comment by greysteil** [comment]
Got you. OK, we'll think on it - thanks for the feedback.

**Comment by waghanza** [comment]
@greysteil if you need help, I'll be glad to contribute (and why not use `crystal` :stuck_out_tongue:)
